pkg/clients/resourcerecordset/resourcerecordset.go

```diff
@@ -0,0 +1,220 @@
+}
+
+// NewClient creates new AWS client with provided AWS Configuration/Credentials
+func NewClient(config *aws.Config) Client {
+	return route53.New(*config)
+}
+
+// GenerateChangeResourceRecordSetsInput prepares input for a ChangeResourceRecordSetsInput
+func GenerateChangeResourceRecordSetsInput(p *v1alpha1.ResourceRecordSetParameters, action route53.ChangeAction) *route53.ChangeResourceRecordSetsInput { // nolint:gocyclo
+	var ttl *int64
+
+	if p.TTL == nil {
+		num := int64(300)
+		ttl = &num
+	} else {
+		ttl = p.TTL
+	}
+
+	resourceRecords := make([]route53.ResourceRecord, len(p.ResourceRecords))
+	for i, r := range p.ResourceRecords {
+		resourceRecords[i] = route53.ResourceRecord{
+			Value: r.Value,
+		}
+	}
+
+	r := &route53.ResourceRecordSet{
+		Name:            p.Name,
+		Type:            route53.RRType(aws.StringValue(p.Type)),
+		TTL:             ttl,
+		ResourceRecords: resourceRecords,
+	}
+
+	if p.AliasTarget != nil {
+		r.AliasTarget = &route53.AliasTarget{
+			DNSName:              awsclients.LateInitializeStringPtr(p.AliasTarget.DNSName, aws.String("")),
+			EvaluateTargetHealth: awsclients.LateInitializeBoolPtr(p.AliasTarget.EvaluateTargetHealth, aws.Bool(false)),
+			HostedZoneId:         awsclients.LateInitializeStringPtr(p.AliasTarget.HostedZoneID, aws.String("")),
+		}
+	}
+
+	if p.GeoLocation != nil {
+		r.GeoLocation = &route53.GeoLocation{
+			ContinentCode:   awsclients.LateInitializeStringPtr(p.GeoLocation.ContinentCode, aws.String("")),
+			CountryCode:     awsclients.LateInitializeStringPtr(p.GeoLocation.CountryCode, aws.String("")),
+			SubdivisionCode: awsclients.LateInitializeStringPtr(p.GeoLocation.SubdivisionCode, aws.String("")),
+		}
+	}
+
+	if p.SetIdentifier != nil {
+		r.SetIdentifier = p.SetIdentifier
+	}
+	if p.Weight != nil {
+		r.Weight = p.Weight
+	}
+	if p.Failover != "" {
+		r.Failover = route53.ResourceRecordSetFailover(p.Failover)
+	}
+	if p.HealthCheckID != nil {
+		r.HealthCheckId = p.HealthCheckID
+	}
+	if p.MultiValueAnswer != nil {
+		r.MultiValueAnswer = p.MultiValueAnswer
+	}
+	if p.Region != "" {
+		r.Region = route53.ResourceRecordSetRegion(p.Region)
+	}
+	if p.TrafficPolicyInstanceID != nil {
+		r.TrafficPolicyInstanceId = p.TrafficPolicyInstanceID
+	}
+
+	return &route53.ChangeResourceRecordSetsInput{
+		HostedZoneId: p.ZoneID,
+		ChangeBatch: &route53.ChangeBatch{
+			Changes: []route53.Change{
+				{
+					Action:            action,
+					ResourceRecordSet: r,
+				},
+			},
+		},
+	}
+}
+
+// IsUpToDate checks if object is up to date
+func IsUpToDate(p v1alpha1.ResourceRecordSetParameters, rrset route53.ResourceRecordSet) (bool, error) {
+	// check for the root "." found in DNS entries and add if not found
+	if !strings.HasSuffix(*p.Name, ".") {
+		p.Name = aws.String(fmt.Sprintf("%s.", *p.Name))
+	}
+	patch, err := CreatePatch(&rrset, &p)
+	if err != nil {
+		return false, err
+	}
+	return cmp.Equal(&v1alpha1.ResourceRecordSetParameters{}, patch), nil
```

Comment:
It needs to skip reference and selector fields.

Reply:
Suggested change has been incorporated.